### PR TITLE
add option to configure environment limits per project

### PR DESCRIPTION
**Pull request by rtprio**
will address #429 

(still work in progress; promotion codepath not yet considered)

### CODE REVIEW

File: services/api/src/dao/project.js [hunk 1]
```diff
@@ -179,7 +179,8 @@ const addProject = ({ sqlClient }) => async (cred, input) => {
         ${input.pullrequests ? ':pullrequests' : '"true"'},
         ${input.production_environment ? ':production_environment' : 'NULL'},
         ${input.auto_idle ? ':auto_idle' : '1'},
-        ${input.storage_calc ? ':storage_calc' : '1'}
+        ${input.storage_calc ? ':storage_calc' : '1'},
+        ${input.environment_limit ? ':environment_limit' : '2'}
```
Comment (Schnitzel) [inline services/api/src/dao/project.js]: this `2` means that the default is a limit of two environments?
I don't think that's a good default, as we probably will run into issues with that.
Let's do the default at 5  :)

**Comment by Schnitzel** [comment]
@rtprio any chance that you can look at the merge conflicts? Thanks!

**Comment by Schnitzel** [comment]
looks like something is failing with the containers locally, maybe run a test locally to see what exactly is failing.

**Review by Schnitzel — CHANGES_REQUESTED** [review]
sorry to jump into this quite late, but could we call it `developmentEnvironmentsLimit`? as the production environments do not count against the limits I want to make it very clear when people are using the API

File: services/api/src/dao/project.js (hunk 1 above)
Comment (rtprio) [reply services/api/src/dao/project.js]: default to 5 now.

**Comment by rtprio** [comment]
This _should_ be ready to merge.

**Pull request merged**